src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -77,6 +77,30 @@ namespace OpenMS
       bool operator()(const MSSpectrum& a, const MSSpectrum& b) const;
     };
 
+    /// Used to remember what subsets in a spectrum are sorted already to allow faster sorting of the spectrum
+    struct Chunk {
+      Size start;
+      Size end;
+      bool is_sorted;
```

Comment:
annotate with e.g. `bool is_sorted; ///< are the elements in [start, end] sorted already?` or similar